packages/react/src/modules/slack/hooks/useSlackChannels.ts

```diff
@@ -1,77 +1,87 @@
 import { useKnockClient } from "@knocklabs/react-core";
-import { useState, useEffect } from "react";
+import { useInfiniteQuery } from "@tanstack/react-query";
+import { useEffect } from "react";
 
-export interface Channel {
+const MAX_COUNT = 1000;
+const LIMIT_PER_PAGE = 200;
+
+export interface SlackChannel {
   name: string;
   id: string;
-  connected: boolean;
+  user: string;
+  is_private: boolean;
+  is_im: boolean;
+  context_team_id: boolean;
 }
 
 export type ContainerObject = {
   objectId: string;
   collection: string;
 };
 
-export function useSlackChannels(
-  tenant: string,
-  connectionsObject: ContainerObject,
-  knockSlackChannelId: string,
-  shouldRefetch: boolean,
-  setErrorMessage: (errorMessage: string | null) => void,
-  errorMessage: string | null,
-): {
-  data: { channels: Channel[] };
+type SlackChannelQueryOptions = {
+  maxCount?: number;
+  limitPerPage?: number;
+  excludeArchived?: boolean;
+  types?: string;
+  teamId?: string;
+};
+
+type UseSlackChannelsProps = {
+  tenant: string;
+  connectionsObject: ContainerObject;
+  queryOptions?: SlackChannelQueryOptions;
+  knockSlackChannelId: string;
+};
+
+type UseSlackChannelOutput = {
+  data: SlackChannel[];
   isLoading: boolean;
-  error: boolean;
-} {
-  const [isLoading, setIsLoading] = useState(true);
-  const [data, setData] = useState<{ channels: Channel[] }>({
-    channels: [],
-  });
+  refetch: () => void;
+};
 
+export function useSlackChannels({
+  tenant,
+  connectionsObject,
+  knockSlackChannelId,
+  queryOptions,
+}: UseSlackChannelsProps): UseSlackChannelOutput {
   const knockClient = useKnockClient();
 
-  useEffect(() => {
-    if (!errorMessage) {
-      const fetchChannels = async () => {
-        try {
-          await knockClient.slack
-            .getChannels({
-              tenant,
-              connectionsObject,
-              knockChannelId: knockSlackChannelId,
-            })
-            .then((res) => {
-              if (res.code === "ERR_BAD_REQUEST") {
-                const message = res.response?.data?.message
-                setData({ channels: [] });
-                setErrorMessage(message);
-                setIsLoading(false);
-                return;
-              }
+  const fetchChannels = ({ pageParam }: { pageParam: string }) => {
+    return knockClient.slack.getChannels({
+      tenant,
+      connectionsObject,
+      knockChannelId: knockSlackChannelId,
+      queryOptions: {
+        ...queryOptions,
+        cursor: pageParam,
+        limit: queryOptions?.limitPerPage || LIMIT_PER_PAGE,
+      },
+    });
+  };
+
+  const { data, isLoading, isFetching, fetchNextPage, hasNextPage, refetch } =
+    useInfiniteQuery({
+      queryKey: ["slackChannels"],
+      queryFn: fetchChannels,
+      initialPageParam: "",
+      getNextPageParam: (lastPage) =>
+        lastPage?.next_cursor === "" ? null : lastPage?.next_cursor,
+    });
```

Comment:
this is an "infiniteQuery" in that it will keep all of the fetched data in state instead of "paging" through and erasing the previous fetch from state on each call.